docs/xdebug.md

```diff
@@ -0,0 +1,68 @@
+
+2. Run the following commands from your `/archipelago-deployment` directory:
+   
+   `docker-compose down` \
+   `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d`
+   
+   This version of `docker-compose up` uses an override file and to specify we want to use the development image for our PHP service. It will rebuild the containers and now you are ready for Part 2.
+
+## Part 2: PHPStorm
+
+1. In PHPStorm, open your `archipelago-deploymenet` project.
+ 
+2. Go to `Preferences > Languages & Frameworks > PHP > Debug`. In this window there is an Xdebug section. Use these settings:
+    - Debug port: `9001`. (do NOT use the default, 9000)
+    - Can accept external connections: yes, select checkbox
+    
+    Your settings should look like this (img). Hit APPLY and OK.
+![Debug](../imgs/xdebug/php-debug.png)    
+
+3. Go to `Preferences > Languages & Frameworks > PHP > Servers`. We will create a new server here. Use these settings:
+    - Name: `docker-debug-server`
+    - Host: `localhost`
+    - Port: `8001`
+    - Use path mappings: yes, select the checkbox
+    - Under project files, select the top-level `archipelago-deployment` directory in the `File/Directory` column.
+    - In the `Absolute path on the server` add `/var/www/html`
+    
+    Hit APPLY and OK and close the window. (img)
+    ![Debug](../imgs/xdebug/servers.png)    
+
+ 
+ 4. Go to `Run > Edit Configurations`. Hit the `+` Button to create a new PHP Remote Debug. Name whatever you want, I called mine `archipelago`. Use these settings:
+    - Filter debug connection by IDE Key: yes, select the checkbox
+    - Server: select `docker-debug-server` from dropdown (we created this in step 3)
+    - IDE Key: `archipelago` (this matches the key set in our container)
+    ![Debug](../imgs/xdebug/edit-configurations.png)    
+
+ 
+ 5. Validate your connection. With  `Run > Edit Configurations` still open, you can hit the link that says "Validate". Use these settings in the following validation window:
+    - Path to create validation script `<your local path>/archipelago-deployment/web`
+    - Url to validation script: `http://localhost:8001`
+    
+    Hit VALIDATE. You should get a series of green check marks. If you get a warning about missing `php.ini` file, that is OK, our file has a different name in the container (`xdebug.ini`) and is still being read correctly.
+    ![Debug](../imgs/xdebug/validate.png)    
+
+ 
+ # Actually Debugging!
+ 1. One last thing. Hit the button (top right bar of PHPStorm) that looks like a telephone, for `Start Listening for PHP Debug Connections`.
+ 
+      ![Debug](../imgs/xdebug/telephone.png) 
+ 2. Now, you can use `Run > Debug` and select the `docker-debug-server` that we created in the previous steps. The debugging console will appear. Happy debugging!
+ 
+ ####Questions? You can ask Marlo at  [mlongley@metro.org](mlongley@metro.org) or post on our [Google Group](https://groups.google.com/forum/#!forum/archipelago-commons)
```

Comment:
jajaja.Nice. IMHO, Let's try to avoid our personal emails, people will always prefer that (of course!!) and you will end answering things over and over and over!